### PR TITLE
sonarqube: Make default config optional

### LINKED ISSUE #18477: 🐛 Bug Report: Sonarqube config failing config check
### 📜 Description

App startup fails when following the config in the readme.

https://github.com/backstage/backstage/tree/master/plugins/sonarqube-backend#example---multiple-global-instance

This config only has the instances values configured and not the default value

```
sonarqube:
  instances:
    - name: default
      baseUrl: https://default-sonarqube.example.com
      apiKey: 123456789abcdef0123456789abcedf012
    - name: specialProject
      baseUrl: https://special-project-sonarqube.example.com
      apiKey: abcdef0123456789abcedf0123456789ab
```

### 👍 Expected behavior

This config should work as expected and pass validation on startup.

This config does actually pass cli validation without erroring, it is only on startup that this seems to be an issue

### 👎 Actual Behavior with Screenshots

It fails the configuration check because `baseUrl` is required in the config.d.ts

```
 Backend failed to start up, Error: Invalid app bundle schema. If this error is unexpected you need to run `yarn build` in the app. If that doesn't help you should make sure your config schema is correct and rebuild the app bundle again. Caused by the following schema error, Error: Config validation failed, Config must have required property 'baseUrl' { missingProperty=baseUrl } at /sonarqube
```

### 👟 Reproduction steps

Add config as above for sonarqube and try to start the app

(it only seems to break on a built version and not in dev mode)

### 📃 Provide the context for the Bug.

_No response_

### 🖥️ Your Environment

_No response_

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by ScottGuymer**
## Hey, I just made a Pull Request!

Closes #18477

Sonarqube allows both "default" config and the configuration of list of instances. A recent change in backstage has started to cause errors on startup if you do no specify the default config, even if you are using the isntances. 

This seems to be because the default values in the config are not shown as optional parameters and are so correctly flagged as no being present.

This change will make the default config optional so that you can use only instance based config. The plugin itself checks to see if you are mixing and matching the values in an unexpected way.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by Parsifal-M — COMMENTED** [review]
Nice!

Thanks for this, strange it causes an issue in 1.15 I would have thought this would be an issue in previous versions also 🤔💭

**Comment by ScottGuymer** [comment]
It would be nice if there was something I could do to get things working again without having to revert but it doesnt seem like there is anything that can turn this behaviour off..

File: .changeset/mighty-pandas-deny.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/plugin-sonarqube': minor
+'@backstage/plugin-sonarqube-backend': minor
```
Comment (freben) [inline .changeset/mighty-pandas-deny.md:3]: These should probably be patch level.

Comment (ScottGuymer) [inline .changeset/mighty-pandas-deny.md:3]:
```suggestion
'@backstage/plugin-sonarqube': patch
'@backstage/plugin-sonarqube-backend': patch
```

Comment (ScottGuymer) [reply .changeset/mighty-pandas-deny.md]: Fixed

**Review by benjdlambert — APPROVED** [review]

**Comment by benjdlambert** [comment]
Nice thanks :tada:

**Pull request merged**